language-design.md

```diff
@@ -16,19 +16,78 @@ are:
+> Our commenter assumes that using JavaScript, as opposed to a “real” language, is a sign of 
+> incapacity. But in technical terms, JavaScript, Perl, and Python are fairly similar (they are
+> all interpreted, dynamically typed, multi-paradigm languages). The only difference is that 
+> our commenter has ideas about who uses each language–and about the languages that “real 
+> programmers” use"
+[Yang and Rabkin, 2015].
+
+This quote is quite accurate in both of our experiences. People stereotype languages based on the users.
+For example the article mentions that Python and Ruby are considered "girly." Both of
+these languages are used a lot in web development, which tends to have a large UI and design
+component. This strong association with creativity, a steretypically feminine quality, means that the languages
+are stereotyped as "girly." On the other hand, C or C++ are seen as manly are often used to solve back-end 
+problems, or find the "right" answer, so they are seen as having less creativity.
+In this way, C and C++ programmers are considered objective in their search for the solution,
+while web development and UI are seen as much more subjective. This gendered differentiation
+hurts everyone who does not conform to these stereotypes.
+
+> "PL/I has not grown much. It is, for the most part, just as it was when it first came
+> out. It may be that this is just from lack of use. The flip side is that the lack of use may
+> have two causes. Number one: PL/I was not designed to grow—it was designed to be all
+> things to all who program right from the start. Number two: for its time, it started out
+> large. No one knew all of PL/I; some said that no one could know all of PL/I"
+[Steele, 1998].
+
+This quote contrasts with the ideas that we discussed in class, specifically that languages cycle
+through developing into a DSL, then into a general purpose language, and then back and through
+the cycle. PL/I was designed to not grow, which is a different philosophy from the other
+languages we have encountered. It would be interesting to discover what drove the design 
+philosophy for PL/I because it is such a large, immutable language.
+
+> "A language design can no longer be a thing. It must be a pattern—a pattern for growth—a 
+> pattern for growing the pattern for defining the patterns that programmers can use for 
+> their real work and their main goal"
+[Steele, 1998].
+
+By thinking of language as a pattern, Steele turns the focus from the programmer to the user.
+By understanding the pattern, users can expand the language, which Steele calls "a pattern for
+growth." At the very least, users feel like they are contributing to an ongoing project as
+opposed to using the results of a finished project. As a bonus, user feedback and suggestions 
+can facilitate a language's growth. The difficulty for programmers is not being able to consider
+what they would like to see, but considering what a user of their language would like to see.
```

Comment:
The user is, as you say, certainly the target programmers should  be thinking about. Something to note is that the way users are able to grow a language is just as important as the ability to grow the language. One thing way to make the users feel like their contributions are "real" is to make their contributions look like things in the language. When the users define new classes and types, having those types look like the primitives in the language is a plus.